core/frontend/src/types/autopilot/parameter-fetcher.ts

```diff
@@ -0,0 +1,105 @@
+import mavlink2rest from '@/libs/MAVLink2Rest'
+// eslint-disable-next-line import/no-cycle
+import { AutopilotStore } from '@/store/autopilot'
+
+import ParametersTable from './parameter-table'
+
+export default class ParameterFetcher {
+  parameter_table = new ParametersTable()
+
+  listener = mavlink2rest.startListening('PARAM_VALUE')
+
+  store = null as (AutopilotStore | null)
+
+  param_count = 0
+
+  total_param_count = null as null | number
+
+  watchdog_last_count = 0
+
+  constructor() {
+    this.setupWs()
+  }
+
+  setStore(store: AutopilotStore): void {
+    this.store = store
+  }
+
+  reset(): void {
+    this.param_count = 0
+    this.total_param_count = null
+    this.watchdog_last_count = 0
+    this.parameter_table.reset()
+  }
+
+  updateStore(): void {
+    if (!this.store) {
+      return
+    }
+    this.store.setParameters(this.parameter_table.parameters())
+    this.store.setParametersCount(this.param_count)
```

Comment:
Looking at the code, this appears to be `parameters_loaded` and `parameters_count` should be `total_param_count`.

Reply:
I updated the code fixing this and renaming stuff so it is clearer